packages/nimble-components/src/wafer-map/modules/prerendering.ts

```diff
@@ -0,0 +1,135 @@
+import { ScaleLinear, scaleLinear, ScaleOrdinal, scaleOrdinal } from 'd3-scale';
+import { WaferMapColorsScaleMode } from '../types';
+import type {
+    Dimensions,
+    Margin,
+    RenderDie,
+    WaferMapDie,
+    WaferMapColorsScale
+} from '../types';
+
+/**
+ * Prerendering prepares render ready dies data  to be used by the rendering module
+ */
+export class Prerendering {
+    public readonly labelsFontSize: number;
+
+    public readonly renderDies: RenderDie[];
+
+    private readonly colorScale:
+    | ScaleOrdinal<string, string>
+    | ScaleLinear<string, string>;
+
+    private readonly fontSizeFactor = 0.8;
+    private readonly nonHighlightedOpacity = 0.3;
+    private readonly emptyDieColor = '#DADFEC';
```

Comment:
At some point it would be good to add these values to `@ni/nimble-tokens` so that they're available in a central location and easy to change.

The instructions are available [in nimble-tokens CONTRIBUTING.md](https://github.com/ni/nimble/blob/main/packages/nimble-tokens/CONTRIBUTING.md). Because this requires coordinating with someone who has Adobe XD access, I'd be ok to track it as follow-up work.

Reply:
We are also in a state where we can't define the tokens in Adobe XD either. I think the right place to capture the tech debt is by adding a note to https://github.com/ni/nimble/issues/781

Reply:
Should I add a note or is this mention enough?

Reply:
Adding a note is helpful because it saves someone reading that issue from having to dig into this PR to get context. No need to worry about it in this case though; I just added a note.

Reply:
Thank you!